app/presenters/corporate_information_page_presenter.rb

```diff
@@ -0,0 +1,69 @@
+class CorporateInformationPagePresenter < ContentItemPresenter
+  include LinkHelper
+
+  def headers_for_contents_list_component
+    return [] unless content_item.headers
+
+    ContentsOutlinePresenter.new(contents_list_headings).for_contents_list_component
+  end
+
+  def corporate_information_heading
+    heading_text = I18n.t("formats.corporate_information_page.corporate_information")
+    heading_id = heading_text.tr(" ", "-").downcase
+
+    {
+      "text" => heading_text,
+      "level" => 2,
+      "id" => heading_id,
+    }
+  end
+
+  def further_information
+    return unless corporate_information_pages?
+
+    further_information_links
+  end
+
+private
+
+  def contents_list_headings
+    content_item.headers << corporate_information_heading if content_item.corporate_information?
+
+    exclude_nested_headings
+
+    ContentsOutline.new(content_item.headers)
```

Comment:
Oh, it's modelled here. Perhaps the modelling should be moved to the model for [consistency](https://github.com/alphagov/frontend/blob/main/app/models/specialist_document.rb#L15)?

Reply:
Ah yes, I did consider this initially, but both the model and presenter uses the method [corporate_information_heading](https://github.com/alphagov/frontend/pull/4608/commits/d45d542fd523fe4af3e71c878232454d9c75c88b#diff-59cebc1a6c28f24f1ed9f80839a0e0a226b94925414b84bfc32fd143d200f356R10) to show the header in the view, and to add it to the `content_item.headers` array.

Reply:
Do you mean the view and the presenter? rather than model and presenter?

It's not a blocker. Just a question.

Reply:
I've moved this to the model now in commit [Model headers for contents list using ContentsOutline model,](https://github.com/alphagov/frontend/pull/4608/commits/db22e38f0cb0c2689901b6016eb91d526cc6a561) as it does make more sense 👍